src/lib/y2partitioner/actions/controllers/filesystem.rb

```diff
@@ -344,21 +405,24 @@ def create_filesystem(type)
         end
 
         def restore_filesystem
+          mount_path = filesystem.mount_path
           mount_by = filesystem.mount_by
-          mount_point = filesystem.mount_point
           label = filesystem.label
 
           @backup_graph.copy(working_graph)
           @backup_graph = nil
           @encrypt = blk_device.encrypted?
 
-          assign_fs_attrs(mount_by: mount_by, label: label)
-          self.mount_point = mount_point
+          filesystem.label = label if label
+
+          restore_mount_point(mount_path, mount_by: mount_by)
         end
 
-        def assign_fs_attrs(attrs = {})
-          attrs.each_pair do |attr, value|
-            filesystem.send(:"#{attr}=", value) unless value.nil?
+        def apply_mount_point_options(options)
```

Comment:
Since it's no longer obvious like in the old code what type "options" is, please document it.